### ISSUE TITLE
Fails to start when using just IP Discovery without a bluetooth adapter

### ISSUE THREAD

**Issue by mod812** [Apollon77/hap-controller-node#202]
**Describe the bug**  
The library works great when you have a bluetooth adapter available, however if just wanting to use IP based discovery without a bluetooth adapter installed, it will fail to start.

**To Reproduce**  
1. Installed Library via NPM.
2. Copied the IP Discovery Sample from the examples directory in the git repo.  
3. Run the sample.  

**Expected behavior**  
IP DIscovery should start without a Bluetooth adapter.

**Screenshots & Logfiles**  
/mnt/c/Users/mod812/source/homehub/node_modules/@abandonware/bluetooth-hci-socket/lib/native.js:9
    super(...args);
    ^

Error: EAFNOSUPPORT, Address family not supported by protocol
    at new BluetoothHciSocketWrapped (/mnt/c/Users/mod812/source/homehub/node_modules/@abandonware/bluetooth-hci-socket/lib/native.js:9:5)
    at new Hci (/mnt/c/Users/mod812/source/homehub/node_modules/@abandonware/noble/lib/hci-socket/hci.js:99:18)
    at new NobleBindings (/mnt/c/Users/mod812/source/homehub/node_modules/@abandonware/noble/lib/hci-socket/bindings.js:27:15)
    at module.exports (/mnt/c/Users/mod812/source/homehub/node_modules/@abandonware/noble/lib/resolve-bindings.js:34:12)
    at module.exports (/mnt/c/Users/mod812/source/homehub/node_modules/@abandonware/noble/with-custom-binding.js:3:53)
    at Object.<anonymous> (/mnt/c/Users/mod812/source/homehub/node_modules/@abandonware/noble/index.js:1:50)
    at Module._compile (node:internal/modules/cjs/loader:1198:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1252:10)
    at Module.load (node:internal/modules/cjs/loader:1076:32)
    at Function.Module._load (node:internal/modules/cjs/loader:911:12) {
  errno: 97,
  code: 'EAFNOSUPPORT',
  syscall: 'socket'
}

**Versions:**  
 - Node version: 20.10.0
 - Operating system:  Ubuntu and Windows 11

**Additional context**  
Tried on a linux box with a bluetooth adapter and it works great.

Thanks,
Derrick

**Comment by Apollon77** [comment]
It seems that ths lazy BLE loading is not functioning correctly .. hmm ... need to look into it

**Comment by NebzHB** [comment]
Dear, I reproduce too (latest version), did you have a look on it ?

**Comment by Apollon77** [comment]
Can you please try if the github version which uses another noble fork works better?

**Comment by NebzHB** [comment]
Hi,

same with latest main branch : 

```
Node.js v20.18.0
/var/www/html/plugins/hkControl/resources/node_modules/hap-controller/node_modules/@stoprocent/bluetooth-hci-socket/lib/native.js:12
class BluetoothHciSocketWrapped extends BluetoothHciSocket {
^
Error: socket creation failed
at new BluetoothHciSocketWrapped (/var/www/html/plugins/hkControl/resources/node_modules/hap-controller/node_modules/@stoprocent/bluetooth-hci-socket/lib/native.js:12:1)
at new Hci (/var/www/html/plugins/hkControl/resources/node_modules/hap-controller/node_modules/@stoprocent/noble/lib/hci-socket/hci.js:100:18)
at new NobleBindings (/var/www/html/plugins/hkControl/resources/node_modules/hap-controller/node_modules/@stoprocent/noble/lib/hci-socket/bindings.js:27:15)
at module.exports (/var/www/html/plugins/hkControl/resources/node_modules/hap-controller/node_modules/@stoprocent/noble/lib/resolve-bindings.js:36:12)
at module.exports (/var/www/html/plugins/hkControl/resources/node_modules/hap-controller/node_modules/@stoprocent/noble/with-custom-binding.js:3:53)
at Object.<anonymous> (/var/www/html/plugins/hkControl/resources/node_modules/hap-controller/node_modules/@stoprocent/noble/index.js:1:50)
at Module._compile (node:internal/modules/cjs/loader:1469:14)
at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
at Module.load (node:internal/modules/cjs/loader:1288:32)
at Module._load (node:internal/modules/cjs/loader:1104:12)
Node.js v20.18.0
```

**Comment by Apollon77** [comment]
@NebzHB Could you please try to put a try/catch around https://github.com/Apollon77/hap-controller-node/blob/b6bd185120a15d24669580bcad03abb84ab0c6b7/src/transport/ble/ble-discovery.ts#L11-L16 and see if this would catch it?

**Comment by NebzHB** [comment]
yes it does

`catched : Error: socket creation failed
at new BluetoothHciSocketWrapped (/var/www/html/plugins/hkControl/resources/node_modules/hap-controller/node_modules/@stoprocent/bluetooth-hci-socket/lib/native.js:12:1)
at new Hci (/var/www/html/plugins/hkControl/resources/node_modules/hap-controller/node_modules/@stoprocent/noble/lib/hci-socket/hci.js:100:18)
at new NobleBindings (/var/www/html/plugins/hkControl/resources/node_modules/hap-controller/node_modules/@stoprocent/noble/lib/hci-socket/bindings.js:27:15)
at module.exports (/var/www/html/plugins/hkControl/resources/node_modules/hap-controller/node_modules/@stoprocent/noble/lib/resolve-bindings.js:36:12)
at module.exports (/var/www/html/plugins/hkControl/resources/node_modules/hap-controller/node_modules/@stoprocent/noble/with-custom-binding.js:3:53)
at Object.<anonymous> (/var/www/html/plugins/hkControl/resources/node_modules/hap-controller/node_modules/@stoprocent/noble/index.js:1:50)
at Module._compile (node:internal/modules/cjs/loader:1469:14)
at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
at Module.load (node:internal/modules/cjs/loader:1288:32)
at Module._load (node:internal/modules/cjs/loader:1104:12)`

**Comment by Apollon77** [comment]
Alright then I know how to handle it. Thank you

**Comment by NebzHB** [comment]
thank you for your swift response !

**Comment by Apollon77** [comment]
Main question now is what should happen? In fact it is only included when the dev thinks ble is used ... so silently not working is mehh ... should it throw exceptions when "start" or such is called? But also then dev can not do too much ... Any idea?

**Comment by NebzHB** [comment]
maybe a setting to disable ble if not needed ?

**Comment by Apollon77** [comment]
Ahhh ok I get it now ... then i can throw exceptions when you try to use it. YOur issue is that you always require it and it throows already then. got it

**Comment by NebzHB** [comment]
i'm not using ble on my project... so if it's disableable it's ok for me ;)

**Comment by Apollon77** [comment]
Ok, I updated github. Because I am limits currently in testing would you mind trying please

**Comment by NebzHB** [comment]
it's working, but i receive the error message. could it be optional ?

**Comment by Apollon77** [comment]
the console log ... I can remove ...

**Comment by Apollon77** [comment]
fixed in 0.10.2 ... on it's way to npm

**Comment by NebzHB** [comment]
perfect ! thanks !
